### PR TITLE
Adding the Features page to the landing

### PR DESCRIPTION
**Pull request by znegrin**
This PR is related to the issue: https://github.com/src-d/backlog/issues/1404

This adds the Features page to the landing:

image

Signed-off-by: Zuri Negrín <zurinegrin@gmail.com>

### CODE REVIEW

**Comment by dpordomingo** [comment]
available at staging
https://landing-staging.srcd.run/features/

got two conflicts during merge, and decided the following:
>~`$medium-light:  #9F9F9F;`~
>`$medium-light:  #989898;`

and
>~`$purple:        #d210c9;`~
>`$purple:        #8719cb;`

File: hugo/layouts/features/single.html [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        <header class="mb-5 text-center">
+            <h2 class="text-dark">Visualize your software development data 
to make better decisions</h2>
+        </header>
+        <div class="row">
+            <div class="col-sm-6 mb-5">
+                <h3 class="text-medium-light mb-4">Transparent roadmap and Ecosystem collaboration</h3>
+                <p class="text-dark">With source{d} you can start getting answers to tough questions in a matter of minutes with simple SQL queries. source{d} supports all major operating systems (Linux, Windows, MacOS), hosting environments (Cloud or On-premise), version control systems (GitHub, GitLab, Bitbucket, etc) and adapts to different stages of development, workflows and business needs. Enjoy the flexibility to choose whether to work on our built-in interactive web interface, your favorite SQL client, command-line interface or Business Intelligence platform (Tableau, Looker, PowerBI, etc).</p>
+                <div class="text-center d-flex mt-5">
+                    <a href="#" class="btn btn-purple text-white flex-fill mr-4">Download source{d}</a>
```
Comment (dpordomingo) [inline hugo/layouts/features/single.html]: broken link

File: hugo/layouts/features/single.html [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                    <a href="#" class="btn btn-highlight flex-fill">Request a Demo</a>
```
Comment (dpordomingo) [inline hugo/layouts/features/single.html]: broken link

File: hugo/layouts/features/single.html [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                <h4 class="title-line line-yellow">Code Retrieval</h4>
+                <p class="mb-5">Retrieve and store the code history of your organization (including your open-source repositories) as a dataset.</p>
+                <h4 class="title-line line-yellow">Analysis in/for any Language</h4>
+                <p class="mb-5">Automatically identify languages, parse source code, and extract the pieces that matter in a distributed and language-agnostic way.</p>
+                <h4 class="title-line line-yellow">History Analysis</h4>
+                <p class="mb-5">Extract information from the evolution, commits, and metadata of your codebase and generate detailed reports and insights.</p>
+                <h4 class="title-line line-yellow">Familiar APIs</h4>
+                <p class="mb-5">Analyze your code through powerful friendly APIs, such as SQL, gRPC, REST, and various client libraries. Use tools you're familiar with to create reports and dashboards.</p>
+                <div class="text-center d-flex mt-5">
+                    <a href="#" class="btn btn-purple text-white mr-4 flex-fill">Watch Video</a>
```
Comment (dpordomingo) [inline hugo/layouts/features/single.html]: do we already have a video?

File: hugo/layouts/features/single.html [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                    <a href="#" class="btn btn-highlight flex-fill">Technical Documentation</a>
```
Comment (dpordomingo) [inline hugo/layouts/features/single.html]: broken link

File: hugo/layouts/features/single.html [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+                <h4 class="title-line line-yellow mt-4">go-git</h4>
+                <p>is a highly extensible git implementation library written in pure Go. It can be used to manipulate git repositories at low level (plumbing) or high level (porcelain), through an idiomatic Go API.</p>
+                <h4 class="title-line line-yellow mt-4 mt-4">Babelfish</h4>
+                <p>Babelfish is a self-hosted server for source code parsing. It can parse any file, in any supported language, extracting an Abstract Syntax Tree (AST) from it and converting it into a Universal Abstract Syntax Tree (UAST).</p>
+            </div>
+            <div class="col-sm-6 mb-5">
+                <img class="img-fluid p-3" src="img/open-components.jpg" alt="">
+            </div>
+            <div class="col-sm-12 d-flex justify-content-center">
+                <a href="#" class="btn btn-highlight-light text-white mr-4">Learn More</a>
```
Comment (dpordomingo) [inline hugo/layouts/features/single.html]: broken link

File: hugo/layouts/features/single.html [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+                <a href="#" class="btn btn-highlight">Join the Community</a>
```
Comment (dpordomingo) [inline hugo/layouts/features/single.html]: broken link

File: hugo/layouts/partials/how-works.html [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+            <h2>How source{d} works for you</h2>
+        </header>
+        <div class="row">
+            <div class="col-sm-4 mb-5">
+                <div class="text-center mb-4">
+                    <img src="img/icons/engineers-and-researchers.svg" alt="Engineers & Researchers" height="150" width="150">
+                </div>
+                <h3 class="title-line line-yellow">Engineers & Researchers</h3>
+                <p class="mb-4">From data retrieval, language analysis and machine learning on code tools, datasets and models, the source{d} stack can be helpful for many software engineering use cases.</p>
+                <a href="get-started" class="btn btn-highlight w-100">Learn More</a>
```
Comment (dpordomingo) [inline hugo/layouts/partials/how-works.html]: The three links point to the very same page and section;
I wonder if it should open its dedicated section depending on your interests: research, management or business.

File: hugo/layouts/features/single.html [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+</section>
+
+<section id="all" class="py-5">
+    <div class="container text-dark">
+        <header class="text-center mb-5">
+            <h2>Get all your code and engineering data in one place</h2>
+        </header>
+        <div class="row">
+            <div class="col-sm-6 mb-5">
+                <img class="img-fluid p-3" src="img/all.svg" alt="">
```
Comment (dpordomingo) [inline hugo/layouts/features/single.html]: This diagram says `source{d} Engine`. I think it should be `source{d}`
image

**Comment by znegrin** [comment]
Just pushed a fix on the buttons so now it looks fine on small mobiles too.

In terms of the links and the diagram, I guess @vcoisne will know better than me what to do in each of the cases.

Thank you!

**Comment by dpordomingo** [comment]
@znegrin problem with viewport in mobile was solved and ready in staging, thanks!

**Comment by znegrin** [comment]
@vcoisne I updated the page. Notice that I couldn't updated these points:

- Update architecture diagram to updated version without Engine mention (I guess there's still not a final design for this, or at least that was my impression from the thread)

- Update the image with source{d} tech stack (the link is broken and shows a 404).

The rest is ready with my latest commit.

image

**Comment by dpordomingo** [comment]
Already in staging:
https://landing-staging.srcd.run/features/

**Comment by dpordomingo** [comment]
About stack image, I think @vcoisne is talking about this image (from the private issue)
https://user-images.githubusercontent.com/5384543/58359832-41e15480-7e3a-11e9-84f0-6249a8e45e62.jpg
source{d} sales deck - May 2019 (1)
(But imo the boxes in the last row, should be vertical aligned, and it should be used a retina version)

File: hugo/layouts/partials/how-works.html (hunk 7 above)
Comment (dpordomingo) [reply hugo/layouts/partials/how-works.html]: imo, these three links should be as done in https://github.com/src-d/landing/pull/405/commits/9f5392c1000fca0a1acdaf3cdbab83d0dd5b952a
- `get-started#engineers-tab`
- `get-started#managers-tab`
- `get-started#executives-tab`

**Comment by vcoisne** [comment]
Thank you @znegrin 

Please make the following changes:

- Update architecture diagram below without Engine mention 
CE
- Replace "Transparent roadmap and Ecosystem collaboration:" at the top of the page by "Built in UI and integrations with your favorite tools
- CTA buttons are too long compared to other pages. Please use the other smaller size across the board. 
- Under “From code retrieval to insights in minutes” remove the following titles “code retrieval” “analysis in/for any language”, History Analysis and “Familiar APIs”. They should be in bold and on the same line as their explanation texts (as is on Balsamiq)
- Under “How sourced works for you: Update as follow
	- Engineers & Researchers to Developers
	- Engineering Manager to Manager
	- Business Leaders to Executives

**Comment by znegrin** [comment]
Hi @vcoisne,

I made the requested changes. Notice that I made the best I could with the buttons to make them look fine responsively, as some of the labels are too long and showing two in the same line breaks in smaller resolutions.

image

**Comment by dpordomingo** [comment]
ready to go, but imo we should use a better image for bblfsh screenshot, that is **low-low-res**
image

**Comment by dpordomingo** [comment]
I'd purpose:
bblfsh

so it would be:
image

**Comment by znegrin** [comment]
Thank you for the image @dpordomingo, I just updated the branch.

image

**Comment by dpordomingo** [comment]
Thanks,
Already in staging:
https://landing-staging.srcd.run/features/

**Comment by vcoisne** [comment]
Thank you @dpordomingo and @znegrin 

Only one more change on that page per Eiso's request. Please replace the architecture diagram by the following: 

how-it-works-two

**Comment by znegrin** [comment]
> Thank you @dpordomingo and @znegrin
> 
> Only one more change on that page per Eiso's request. Please replace the architecture diagram by the following:
> 
> how-it-works-two

It seems that the image resolution is not the optimal. I changed it anyway, but if you have al alternative that looks less blurry it would be great.

**Comment by ricardobaeta** [comment]
Hi @znegrin ! I believe this one is better.

how-it-works-two

**Comment by znegrin** [comment]
As requested on #405 I updated the page to source{d} Community Edition ([products/community-edition](https://sourced.tech/products/community-edition))

**Comment by znegrin** [comment]
I updated the image to the one that @ricardobaeta shared on https://github.com/src-d/landing/pull/399#issuecomment-505553807

**Comment by dpordomingo** [comment]
Is this ready to be merged @vcoisne ?

**Comment by ricardobaeta** [comment]
@zurk @dpordomingo I've updated the image I've provided on [my comment](https://github.com/src-d/landing/pull/399#issuecomment-505553807). 

Please consider the new one:
how-it-works-two

**Comment by znegrin** [comment]
> @Zuri please have in mind my [latest comment](https://github.com/src-d/landing/pull/399#issuecomment-505622197). Thank you.

It is now updated. Thanks.

**Pull request merged**